aiomusiccast/pyamaha.py

```diff
@@ -1035,9 +1035,12 @@ def set_volume(zone, volume, step):
                     Values: Value range calculated by minimum/maximum/step values gotten via /system/getFeatures.
         """
         assert zone in ZONES, 'Invalid ZONE value!'
-        return Zone.URI['SET_VOLUME'].format(
-            host='{host}', zone=zone, volume=volume, step=step
+        url = Zone.URI['SET_VOLUME'].format(
+            host='{host}', zone=zone, volume=volume
         )
+        if step:
+            url += f"&step={step}"
+        return url
```

Comment:
This change is not necessary to get the new volume up/down functions working, right? You just made the adjustments to have a clean URL, if the value is set to None. I tried it without these changes and it was still working as expected. Maybe we should implement our own URL formatting method and only return URLs, which do not contain None as values.